src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -263,6 +263,12 @@ object Erasure extends TypeTestsCasts{
           else
             cast(tree, pt)
       }
+
+
+    /** Is `tpe` a type which is a bottom type for Dotty, but not a bottom type for JVM */
+    def isNonJVMBottomType(tpe: Type)(implicit ctx: Context): Boolean = {
+      tpe.derivesFrom(ctx.definitions.NothingClass) || tpe.derivesFrom(ctx.definitions.NullClass)
+    }
```

Comment:
I now did a similar change in definitions (f780a371a3716c6917a510ed72a5927704634bb3). 
So it's best if you cherry-pick the commit and use defn.isBottomType.